src/parser.js

```diff
@@ -267,8 +267,16 @@ export default class Parser extends Stream {
               this.manifest.discontinuityStarts.push(uris.length);
             },
             'program-date-time'() {
-              this.manifest.dateTimeString = entry.dateTimeString;
-              this.manifest.dateTimeObject = entry.dateTimeObject;
+              if (typeof this.manifest.dateTimeString === 'undefined') {
+                // PROGRAM-DATE-TIME is a media-segment tag, but for backwards
+                // compatibility, we add the first occurance of the PROGRAM-DATE-TIME tag
+                // to the manifest object
+                this.manifest.dateTimeString = entry.dateTimeString;
+                this.manifest.dateTimeObject = entry.dateTimeObject;
+              }
```

Comment:
This is needed for calculating stream time from currentTime, so is more important than "for backwards compatibility" implies.